lib/MPIPreferences/src/parse_cray_cc.jl

```diff
@@ -0,0 +1,94 @@
+module CrayParser
+
+import Base: filter, map, reduce
+filter(f::Function)::Function = x -> filter(f, x)
+map(f::Function)::Function    = x -> map(f, x)
+reduce(f::Function)::Function = x -> reduce(f, x)
+
+struct CrayPE
+    libmpi::String
+    libgtl::String
+    cclibs::Vector{String}
+    gtl_env_switch::String
+    
+    CrayPE(mpi_dl::T, gtl_dl::T, cclibs::Vector{T}) where T <:AbstractString = new(
+        "lib" * mpi_dl * ".so", # Assuming Linux -- CrayPE is only avaialbe for linux anyway
+        "lib" * gtl_dl * ".so",
+        cclibs,
+        "MPICH_GPU_SUPPORT_ENABLED"
+    )
+end
+
+function communicate(cmd::Cmd, input="")
+    inp = Pipe()
+    out = Pipe()
+    err = Pipe()
+
+    process = run(pipeline(cmd, stdin=inp, stdout=out, stderr=err), wait=false)
+
+    close(out.in)
+    close(err.in)
+
+    stdout = @async String(read(out))
+    stderr = @async String(read(err))
+
+    write(process, input)
+    close(inp)
+
+    wait(process)
+    return (
+        stdout = fetch(stdout),
+        stderr = fetch(stderr),
+        code = process.exitcode
+    )
+end
+
+const libmpi_prefix = "mpi_"
+const libgtl_prefix = "mpi_gtl_"
+
+function cray_mpi(libs)
+    x = libs |>
+        filter(x-> startswith(x, libmpi_prefix)) |> 
+        filter(x->!startswith(x, libgtl_prefix))
```

Comment:
For example this can be simplified to
```suggestion
        Base.Fix1(filter)(startswith(libmpi_prefix)) |> 
        Base.Fix1(filter)(!startswith(libgtl_prefix))
```

Reply:
I've just defined `filter` etc at the top of the file -- without importing Base.filter, etc first -- I think it makes the code more readable